### PR TITLE
WIP Support pokestop webhook messages and incident alerts

### PR DESCRIPTION
**Pull request by fosJoddie**
### WARNING
This PR will require you to change your DTS with new stuff, check out "incident" section in the updated dts examples.

## Description
Add support for receiving pokestop webhooks and handle the incidents that come there. 
Team Rocket Incidents can be tracked with `!incident` command in a channel or PM, and will either react to incidents in your area(s) or based on distance from location if distance has been set like`!incident d5000`.

DTS names are `name` for pokestop name and `imgurl` for pokestop url. you have your normal tth and time for the EXPIRATION/END time and thats pretty much it. We do not know reward or anything like that.

For now there is ONLY support for MAD messages, since i do not have a RDM setup. If someone send me the payload for RDM I'll retrofit that in. Probably.

There's no way to see what incidents you track with `!tracked` - it has to be added in later.

## Motivation and Context
Because I wanna beat Team Rocket

## How Has This Been Tested?
Limited local testing on white bunnies and yellow chickens.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: config/examples/dts.json.example [hunk 1]
```diff
@@ -302,6 +302,59 @@
...
+        "description": "[Google Maps]({{{mapurl}}}) [Apple Maps]({{{applemap}}})",
+        "author": {
+          "name": "end: {{time}} in {{tthm}}m {{tths}}s",
+          "icon_url": "{{{imgurl}}}"
+        }
+      }
+    },
+    "3": {
+      "embed": {
+        "title": "Team Rocket incident has started! at {{ame}}",
```
Comment (Kneckter) [inline config/examples/dts.json.example]: Typo "ame" should be "name". Just a small fix for anyone testing.

File: src/controllers/incident.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+				}
+				case 'mapbox': {
+					data.staticmap = `https://api.mapbox.com/styles/v1/mapbox/streets-v10/static/url-https%3A%2F%2Fi.imgur.com%2FMK4NUzI.png(${data.longitude},${data.latitude})/${data.longitude},${data.latitude},${config.geocoding.zoom},0,0/${config.geocoding.width}x${config.geocoding.height}?access_token=${_.sample(config.geocoding.staticKey)}`
+					break
+				}
+				default: {
+					data.staticmap = ''
+				}
+			}
+            if(data.name) data.gym_name = data.name;
```
Comment (Kneckter) [inline src/controllers/incident.js]: I don't think this is used since we are not using gym_name.

File: src/controllers/incident.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+			let incidentExpiration = data.incident_expiration ? data.incident_expiration : data.incident_expire_timestamp;
+			data.tth = moment.preciseDiff(Date.now(), incidentExpiration * 1000, true)
+			data.distime = moment(incidentExpiration * 1000).tz(geoTz(data.latitude, data.longitude).toString()).format(config.locale.time)
+			
+			// Stop handling if it already disappeared or is about to go away
+			if (data.tth.firstDateWasLater || ((data.tth.hours * 3600) + (data.tth.minutes * 60) + data.tth.seconds) < minTth) {
+				log.warn(`${data.name} Incident already disappeared or is about to go away in: ${data.tth.hours}:${data.tth.minutes}:${data.tth.seconds}`)
+				resolve([])
+				return null
+			}
```
Comment (Kneckter) [inline src/controllers/incident.js]: We are not passing the TTH so we should not need this check. We can do that NOW() check in the other file like your TODO had.

**Comment by Kneckter** [comment]
Besides the above comments, I tested this with RDM and MAD formats manually and it worked for both.

File: src/controllers/incident.js (hunk 3 above)
Comment (fosJoddie) [reply src/controllers/incident.js]: tth is calculated a few lines above, so this code should prevent any late discoveries, just like with pokemons.

**Comment by fosJoddie** [comment]
Fixed the issues you mentioned @Kneckter , thanks for reporting. 

I gotta setup my own test instance and look for that error. Does it happen to you all the time while testing or only sometimes?

**Comment by Kneckter** [comment]
@fosJoddie the error looks like it was just a me-issue related to the limit on inotify. https://github.com/guard/listen/wiki/Increasing-the-amount-of-inotify-watchers#the-technical-details

I did not get it after in creasing the limit. I tested your updates and did not have an issue using manual data

**Comment by KartulUdus** [comment]
`There's no way to see what incidents you track with !tracked - it has to be added in later.`

I don't see a point to add it to !tracked unless monster is in the tracking

**Comment by Kneckter** [comment]
The `!incident remove` command is not removing DB entries

**Comment by Kneckter** [comment]
I tested out the `!invasion` and `!invasion remove` commands and they worked. This was with live RDM data.

**Comment by ttanguay** [comment]
The gruntType field needs to be NOT NULL since it's included in a PRIMARY KEY... in the db migration5 file.

**Pull request merged**